pages/Identity/Registry.md

```diff
@@ -0,0 +1,173 @@
+
+## Discovery
+
+Contract addresses are not easy to remember and while some networks have had success with just numbers,
+it is most common to think of text based handles for finding other people on a network.
+DSNP handles are an easy way to allow easy user lookup.
+
+## Homograph Attack Mitigation
+
+UTF-8 support for handles opens handle users up to [homograph attacks](https://en.wikipedia.org/wiki/IDN_homograph_attack), not to mention case-sensitivity issues.
+This issue is of ongoing discussion both for the DSNP as well as in for ICANN domain names and other projects working with internationalization support.
+
+Because the DSNP Id is stable, attacks would only be successful in cases where the DSNP Id were unknown.
+[Punycode](https://en.wikipedia.org/wiki/Punycode) would resolve some issues, but would require losing true internationalization.
+
+### Current Mitigation Strategies
+
+Clients resolving handles MUST implement a method to detect potential homographs and check both user settings and potentially check the registry for additional potential matching DSNP Ids.
+
+## DSNP Ids
+
+Ethereum contract addresses are currently 160 bit values which is much larger than needed for unique identification.
+Graph connections can be reduced to just 64 bit identifiers.
+
+**Remember:** Only DSNP Ids are safe for long term data connections.
+
+* DSNP Identity contracts are not guaranteed long term for upgrade and migration reasons
+* Handles are subject to homograph attacks
+
+
+## EIP 721
+
+While the NFT standard is not necessary to support, it provides standard ownership transfer and management tools.
+
+TODO: Research the possibilities of EIP 721 support.
+
+## Rejected Alternatives
+
+* Why not the Ethereum Name Service ([ENS](https://ens.domains/))?
+  * Overly complex and expensive for the simple use case of handles
+  * Support can be added later through a custom ENS resolver
+  * Didn't provide a numerical id for efficient graph storage
+  * Still subject to homograph attacks
+
+## Contract Interface
+
+```solidity
+/**
+ * @dev DSNP Registry Interface
+ */
+interface IRegistry {
+    
+    /**
+     * @dev Log when a new handle is registered
+     * @param addr The address the handle is pointing at
+     * @param id The numerical DSNP Id for storing references
+     * @param handle The actual UTF-8 string used for the handle 
+     */
+    event DSNPHandle(address addr, uint64 id, string handle);
+
+    /**
+     * @dev Log when a the handle address is changed
+     * @param addr The new address the handle is pointing at
+     * @param id The numerical DSNP Id for reference 
+     */
+    event DSNPHandleChange(address addr, uint64 id);
+
+    /**
+     * @dev Register a new handle
+     * @param addr Address for the handle to point at
+     * @param handle The handle for the address
+     * 
+     * MUST NOT allow a registration of an addr that is already attached to a different handle
```

Comment:
If handles are primarily for discovery, do we care if an identity delegation contract is attached to more than one handle?

Let's take the case where someone has two twitter accounts. They don't care that people know the same person is behind both accounts, merely that they have different "accounts" for different content.

This is different than the question of can two handles point to the SAME account.

It comes down to the question of "what is an account?". Is the account the identifier (the DSNP Id) or the ownership?

Reply:
The thinking of the DSNP Id as the "account" and the contract as merely ownership and other abilities means there is no reason to require two handle pointing at the same contract address (albeit suggested to keep separate)

Never however can more than one handle be pointed at the same DSNP Id.